lib/Dialect/Arc/Transforms/ArcCanonicalizer.cpp

```diff
@@ -657,6 +663,46 @@ MergeVectorizeOps::matchAndRewrite(VectorizeOp vecOp,
   return success();
 }
 
+LogicalResult KeepOneVecOp::matchAndRewrite(VectorizeOp vecOp,
+                                            PatternRewriter &rewriter) const {
+  BitVector argsToRemove(vecOp.getInputs().size(), false);
+  DenseMap<VectorizeOp, SmallVector<unsigned>> opExist;
+  for (auto [argIdx, inputVec] : llvm::enumerate(vecOp.getInputs())) {
+    auto otherVecOp = inputVec[0].getDefiningOp<VectorizeOp>();
+    if (!otherVecOp || inputVec != otherVecOp.getResults())
+      continue;
+
+    if (opExist.count(otherVecOp))
+      argsToRemove.set(argIdx);
+
+    opExist[otherVecOp].push_back(argIdx);
+  }
+
+  if (argsToRemove.none())
+    return failure();
+
+  auto &currentBlock = vecOp.getBody().front();
+  for (auto [op, positions] : opExist)
+    for (unsigned idx = 1; idx < positions.size(); ++idx)
+      rewriter.replaceAllUsesWith(currentBlock.getArgument(positions[idx]),
+                                  currentBlock.getArgument(positions[0]));
+
+  currentBlock.eraseArguments(argsToRemove);
+  SmallVector<Value> newOperands;
+  for (auto [argIdx, inputVec] : llvm::enumerate(vecOp.getInputs()))
+    if (!argsToRemove[argIdx])
+      newOperands.insert(newOperands.end(), inputVec.begin(), inputVec.end());
```

Comment:
Nit: we could already build this new operand list during the first iteration over the inputs (instead of having `argsToRemove`)